sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -587,7 +587,8 @@ case class DataSource(
   private def disallowWritingIntervals(
       dataTypes: Seq[DataType],
       forbidAnsiIntervals: Boolean): Unit = {
-    val isWriteAllowedSource = writeAllowedSources(providingClass)
+    val isWriteAllowedSource = writeAllowedSources(providingClass) ||
+      providingClass.getCanonicalName == "org.apache.spark.sql.avro.AvroFileFormat"
```

Comment:
I cannot refer to the `AvroFileFormat` class from `external` at here.

Reply:
I think we can remove this check since we support ANSI interval in all built-in file based datasources. The `text` and `libsvm` must have their own checks.

Reply:
I agree. will you remove this check within  this PR?

Reply:
Yea let's clean it up in this PR, as we are probably done with supporting ANSI intervals in builtin file sources.